File: 2144-maximum-difference-between-increasing-elements/maximum-difference-between-increasing-elements.py
class Solution:
    def maximumDifference(self, nums: List[int]) -> int:
        n = len(nums)
        ans = -1
        mini = inf
        for i in range(n):
            mini = min(mini, nums[i])
            if(nums[i]>mini):
                ans = max(ans, nums[i]-mini)
        
        return ans
        

File: 2144-maximum-difference-between-increasing-elements/maximum-difference-between-increasing-elements.cpp
class Solution {
public:
    int maximumDifference(vector<int>& nums) {
        int ans = -1, n = nums.size(), mini = INT_MAX;
        for(int i = 0; i<n; i++){
            mini = min(mini, nums[i]);
            if(nums[i]>mini) ans = max(ans,nums[i]-mini);
        }
        return ans;
    }
};